pkg/apiserver/api_installer.go

```diff
@@ -567,6 +571,24 @@ func (a *APIInstaller) registerResourceHandlers(path string, storage rest.Storag
 			}
 			addParams(route, action.Params)
 			ws.Route(route)
+		case "DELETECOLLECTION":
+			doc := "delete collection of " + kind
+			if hasSubresource {
+				doc = "delete collection of " + subresource + " of a " + kind
```

Comment:
We have the "hasSubresource" code in pure "DELETE" path:
https://github.com/kubernetes/kubernetes/blob/master/pkg/apiserver/api_installer.go#L554

Since DeleteCollection is logicall a "collection of deletions", I think it makes sense to have it here.